### PR TITLE
feat : Cleanup L1 Genesis Generation and removed unused crypto dependencies

### LINKED ISSUE #12518: op-chain-ops: Cleanup L1 Genesis Generation
We still have codepaths to generate an L1 chainspec that configures clique. The clique engine is no longer maintained as part of `go-ethereum`, we should remove support for it. Once we shipped Ecotone as part of the OP Stack, we enabled blob transactions (EIP-4844). geth running clique does not support blobs, which we generally want to use as part of a local devnet.

This ticket can be considered complete when the following codepath is removed:

https://github.com/ethereum-optimism/optimism/blob/27c12a132d3238b8dda4ea73d947fdd46677a712/op-chain-ops/genesis/genesis.go#L156-L163

As well as removing the `L1UseClique` and `CliqueSignerAddress` fields from the `DeployConfig`

### PR DESCRIPTION
**Pull request by mahmudsudo**
**Description**
Cleanup L1 Genesis Generation by removing clique support

**Tests**

There was no need for additional tests as this was a refactor
all  former tests passes

-closes #12518

### CODE REVIEW

**Comment by tynes** [comment]
Can you please also remove `L1UseClique` from the `DeployConfig` and all additional references to it?

```shell
$ git grep -rin l1useclique

op-chain-ops/genesis/config.go:786:	// L1UseClique represents whether or not to use the clique consensus engine.
op-chain-ops/genesis/config.go:787:	L1UseClique bool `json:"l1UseClique"`
op-chain-ops/genesis/genesis.go:149:	if config.L1UseClique {
op-chain-ops/genesis/genesis.go:181:	if !config.L1UseClique && config.L1CancunTimeOffset != nil {
op-chain-ops/genesis/testdata/test-deploy-config-full.json:9:  "l1UseClique": false,
op-e2e/config/init.go:206:	dc.L1UseClique = false
packages/contracts-bedrock/deploy-config/sepolia-devnet-0.json:19:  "l1UseClique": false,
```

Also the `CliqueSignerAddress`

```shell
op-chain-ops/genesis/config.go:783:	// CliqueSignerAddress represents the signer address for the clique consensus engine.
op-chain-ops/genesis/config.go:785:	CliqueSignerAddress common.Address `json:"cliqueSignerAddress"`
op-chain-ops/genesis/genesis.go:156:		extraData = append(append(make([]byte, 32), config.CliqueSignerAddress[:]...), make([]byte, crypto.SignatureLength)...)
op-chain-ops/genesis/testdata/test-deploy-config-full.json:10:  "cliqueSignerAddress": "0x0000000000000000000000000000000000000000",
packages/contracts-bedrock/deploy-config/sepolia-devnet-0.json:18:  "cliqueSignerAddress": "0x0000000000000000000000000000000000000000",
```

**Comment by mahmudsudo** [comment]
> Can you please also remove `L1UseClique` from the `DeployConfig` and all additional references to it?
> 
> ```shell
> $ git grep -rin l1useclique
> 
> op-chain-ops/genesis/config.go:786:	// L1UseClique represents whether or not to use the clique consensus engine.
> op-chain-ops/genesis/config.go:787:	L1UseClique bool `json:"l1UseClique"`
> op-chain-ops/genesis/genesis.go:149:	if config.L1UseClique {
> op-chain-ops/genesis/genesis.go:181:	if !config.L1UseClique && config.L1CancunTimeOffset != nil {
> op-chain-ops/genesis/testdata/test-deploy-config-full.json:9:  "l1UseClique": false,
> op-e2e/config/init.go:206:	dc.L1UseClique = false
> packages/contracts-bedrock/deploy-config/sepolia-devnet-0.json:19:  "l1UseClique": false,
> ```
> 
> Also the `CliqueSignerAddress`
> 
> ```shell
> op-chain-ops/genesis/config.go:783:	// CliqueSignerAddress represents the signer address for the clique consensus engine.
> op-chain-ops/genesis/config.go:785:	CliqueSignerAddress common.Address `json:"cliqueSignerAddress"`
> op-chain-ops/genesis/genesis.go:156:		extraData = append(append(make([]byte, 32), config.CliqueSignerAddress[:]...), make([]byte, crypto.SignatureLength)...)
> op-chain-ops/genesis/testdata/test-deploy-config-full.json:10:  "cliqueSignerAddress": "0x0000000000000000000000000000000000000000",
> packages/contracts-bedrock/deploy-config/sepolia-devnet-0.json:18:  "cliqueSignerAddress": "0x0000000000000000000000000000000000000000",
> ```

made the necessary adjustment

**Comment by mahmudsudo** [comment]
@tynes

**Comment by tynes** [comment]
CI is currently failing

```
golangci-lint run -E goimports,sqlclosecheck,bodyclose,asciicheck,misspell,errorlint --timeout 5m -e "errors.As" -e "errors.Is" ./...
op-chain-ops/genesis/genesis.go:149: File is not `goimports`-ed (goimports)

op-e2e/config/init.go:204: File is not `goimports`-ed (goimports)
```

**Comment by mahmudsudo** [comment]
@tynes

**Comment by tynes** [comment]
Given the failure in CI, looks like we need to make a small modification to make it pass

This is the error:

```
                Error Trace:    /var/opt/circleci/data/workdir/op-e2e/actions/helpers/l1_replica.go:87
                                                        /var/opt/circleci/data/workdir/op-e2e/actions/helpers/l1_miner.go:49
                                                        /var/opt/circleci/data/workdir/op-e2e/actions/helpers/setups.go:24
                                                        /var/opt/circleci/data/workdir/op-e2e/actions/helpers/setups.go:67
                                                        /var/opt/circleci/data/workdir/op-e2e/actions/upgrades/ecotone_fork_test.go:60
                Error:          Received unexpected error:
                                only PoS networks are supported, please transition old ones with Geth v1.13.x
                Test:           TestEcotoneNetworkUpgradeTransactions
```

You can see how the error is created here: https://github.com/ethereum/go-ethereum/blob/3e567b8b2901611f004b5a6070a9b6d286be128d/eth/ethconfig/config.go#L165

This means we should make a modification around here in the callstack:
https://github.com/ethereum-optimism/optimism/blob/5f2cc843fde7960cb2bc907542b528260a3f5f6f/op-e2e/actions/helpers/l1_replica.go#L54
by adding the `TerminalTotalDifficulty` to the chain config, it should be fine if it is `big.NewInt(0)`, it just cannot be `nil`. There may be a few other places where this will be needed to be added

File: op-chain-ops/genesis/genesis.go [hunk 1]
```diff
@@ -146,21 +145,6 @@ func NewL1Genesis(config *DeployConfig) (*core.Genesis, error) {
...
-			Period: config.L1BlockTime,
-			Epoch:  30000,
-		}
-		extraData = append(append(make([]byte, 32), config.CliqueSignerAddress[:]...), make([]byte, crypto.SignatureLength)...)
-	} else {
-		chainConfig.MergeNetsplitBlock = big.NewInt(0)
-		chainConfig.TerminalTotalDifficulty = big.NewInt(0)
-		chainConfig.TerminalTotalDifficultyPassed = true
-		chainConfig.ShanghaiTime = u64ptr(0)
-		chainConfig.CancunTime = u64ptr(0)
```
Comment (protolambda) [inline op-chain-ops/genesis/genesis.go:162]: This part is important to keep to fix the error.

**Comment by protolambda** [comment]
Closing in favor of https://github.com/ethereum-optimism/optimism/pull/12977

**Pull request closed without merging**